flask_resty/view.py

```diff
@@ -400,9 +400,12 @@ def request_args(self):
                 # KeyError for args that aren't present.
                 continue
 
-            value = args.getlist(field_name)
-            if not isinstance(field, fields.List) and len(value) == 1:
-                value = value[0]
+            if not isinstance(field, fields.List):
+                value = args.get(field_name)
+            else:
+                tmp, value = args.getlist(field_name), []
```

Comment:
the most pythonic way to do this is probably as a single list comprehension

```python
In [2]: a = ['a,b,c', 'd,e']

In [3]: [v2 for v1 in a for v2 in v1.split(',')]
Out[3]: ['a', 'b', 'c', 'd', 'e']
```

Reply:
Or `chain`?

```python
In [1]: from itertools import chain

In [2]: a = ['a,b,c', 'd,e']

In [3]: a_lists = [each.split(',') for each in a]

In [4]: list(chain.from_iterable(a_lists))
Out[4]: ['a', 'b', 'c', 'd', 'e']
```

No strong preference here.

Reply:
Or, to avoid the intermediate list:

```python

In [1]: from itertools import chain

In [2]: a = ['a,b,c', 'd,e']

In [3]: list(chain(*(each.split(',') for each in a)))
Out[3]: ['a', 'b', 'c', 'd', 'e']
```

which I think reads pretty nicely. Still no strong preference between Jimmy's and this approach. @stanley98yu takeaway here is that comprehensions are generally preferred over `for` loops with mutation.